### PR TITLE
Feature/contact details

### PR DESCRIPTION
**Pull request by ztolley**
Added the ability to view a contacts details.
Refactored some of the formatting code into services
Made the phone number field consistent.
Removed redundant contact and interaction list styles.

### CODE REVIEW

File: src/services/contactformattingservice.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+function getDisplayContact (contact) {
+  return {
+    title: (contact.title && contact.title.name) ? contact.title.name : null,
+    job_title: contact.job_title,
+    telephone_number: formatPhone(contact.telephone_countrycode, contact.telephone_number),
+    email: contact.email,
+    address: getFormattedAddress(contact),
+    telephone_alternative: contact.telephone_alternative,
+    email_alternative: contact.email_alternative,
+    notes: newlineToBr(contact.notes)
```
Comment (reupen) [inline src/services/contactformattingservice.js]: I had this fail on a contact where `contact.notes` was null.

File: src/services/contactformattingservice.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+/**
+ * Translate a raw contact object into a formatted contact
+ * to display on the screen
+ * @param {object} contact
+ * @returns {object} displayContact A contact that can be put into a key value table
+ *
+ */
+function getDisplayContact (contact) {
+  return {
+    title: (contact.title && contact.title.name) ? contact.title.name : null,
```
Comment (reupen) [inline src/services/contactformattingservice.js]: We weren't going to display title for the time being (one of the tasks in the sprint).

File: src/lib/textformatting.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+/**
+ * Translate text to sentance case, e.g. the cat sat -> The cat sat
```
Comment (reupen) [inline src/lib/textformatting.js]: Typo in 'sentance'.

File: src/lib/textformatting.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * @param {string} source text
+ * @returns {string} result formatted text
+ */
+function sentenceCase (str) {
```
Comment (reupen) [inline src/lib/textformatting.js]: Could do with some tests – seems like it is title case rather than sentence case?

Comment (ztolley) [inline src/lib/textformatting.js]: Does seem that way. I'll change the name

File: src/repositorys/servicedeliveryrepository.js [hunk 5]
```diff
@@ -18,15 +19,42 @@ function saveServiceDelivery (token, serviceDelivery) {
...
+    })
+  })
 }
 
 function getServiceDeliverysForContact (token, companyId) {
-  return authorisedRequest(token, `${config.apiRoot}/v2/service-delivery/?contact=${companyId}`)
+  return new Promise((resolve) => {
+    authorisedRequest(token, `${config.apiRoot}/v2/service-delivery/?contact=${companyId}`)
+    .then((response) => {
+      resolve(response.dat)
```
Comment (jim68000) [inline src/repositorys/servicedeliveryrepository.js]: response.data?

File: src/views/contact/index.html [hunk 6]
```diff
@@ -1,15 +1,11 @@
...
-</script>
-{% endblock %}
+{% import "macros/trade.html" as trade %}
 
 {% block main %}
-<h1 class="visually-hidden">Contact Details</h1>
-<h2 class="page-heading">{{contact.first_name}} {{contact.last_name}}</h2>
-{{ contact | log | safe }}
+<h1 class="page-heading">
+  {{contact.first_name}} {{contact.last_name}} {% if contact.primary == true %}{{trade.statusbadge('Primary', variation="fuschia")}}{% endif %}
```
Comment (jim68000) [inline src/views/contact/index.html]: FYI, fuchsia is the right spelling.

File: src/repositorys/servicedeliveryrepository.js (hunk 5 above)
Comment (ztolley) [inline src/repositorys/servicedeliveryrepository.js]: nice spot

File: src/views/contact/index.html (hunk 6 above)
Comment (ztolley) [inline src/views/contact/index.html]: i know.. so change it in trade elements...

File: src/services/contactformattingservice.js (hunk 1 above)
Comment (ztolley) [inline src/services/contactformattingservice.js]: The tests for text formatting should cover that scenario now

**Pull request merged**